allennlp/modules/transformer/bimodal_encoder.py

```diff
@@ -0,0 +1,303 @@
+        pretrained_module: torch.nn.Module,
+        source="huggingface",
+        mapping: Optional[Dict[str, str]] = None,
+    ):
+        """
+        The `pretrained_module` only supplies one of the modalities.
+        """
+        submodules = cls._get_mapped_submodules(pretrained_module, source, mapping)
+
+        kwargs = {}
+
+        kwargs["num_hidden_layers1"] = len(submodules["layers1"])
+
+        kwargs["hidden_size1"] = submodules["layers1.0.attention.self.query"].in_features
+        kwargs["num_attention_heads"] = submodules["layers1.0.attention.self"].num_attention_heads
+        kwargs["attention_dropout1"] = submodules["layers1.0.attention.self.dropout"].p
+        kwargs["hidden_dropout1"] = submodules["layers1.0.attention.output.dropout"].p
+        kwargs["intermediate_size1"] = submodules["layers1.0.intermediate.dense"].out_features
+        kwargs["activation"] = submodules["layers1.0.intermediate"].intermediate_act_fn
+
+        return kwargs
+
+    def _load_from_pretrained_module(
+        self,
+        pretrained_module: torch.nn.Module,
+        source="huggingface",
+        mapping: Optional[Dict[str, str]] = None,
+    ):
+        """
+        Loads the weights of the `pretrained_module` into the instance.
+        Optionally, a `mapping` is specified for any differences in parameter names
+        between `pretrained_module` and the instance.
+        """
+        # FIX: ignore_absent_parameters should be a general option.
+        if mapping is None:
+            self._construct_default_mapping(source)
+            mapping = self._default_mapping
+        pretrained_parameters = dict(pretrained_module.named_parameters())
+        ignore_absent_parameters = ["layers2", "c_layer"]  # FIX: specific to source.
+        for name, parameter in self.named_parameters():
+            pretrained_name = name
+            for key, val in mapping.items():
+                # so that we replace the names of submodules too.
+                # eg. module.key.anothermodule --> module.val.anothermodule
+                pretrained_name = pretrained_name.replace(key, val)
+
+            if not any(
+                [pretrained_name.startswith(paraname) for paraname in ignore_absent_parameters]
+            ):
+                if pretrained_name not in pretrained_parameters:
+                    raise ValueError(
+                        f"Couldn't find a matching parameter for {name}. Is this module "
+                        "compatible with the pretrained module you're using?"
+                    )
+                parameter.data.copy_(pretrained_parameters[pretrained_name].data)
+
+    @classmethod
+    def from_pretrained_module(
+        cls,
+        pretrained_module: torch.nn.Module,
+        source="huggingface",
+        mapping: Optional[Dict[str, str]] = None,
+        **kwargs,
+    ):
+        """
+        The `pretrained_module` only supplies one of the modalities.
+        """
+
+        required_kwargs = [
```

Comment:
I don't really understand why we need these arguments to be hidden in `kwargs`.  `kwargs` are really messy and annoying to deal with in general, and I'd strongly recommend avoiding using them except in a few specific use cases.  They make it so you have to look at the method implementation to know what arguments a function takes, instead of just looking at the function definition, and that makes IDE integrations and documentation generators not work.

Is there another solution to the problem this is solving that doesn't require `kwargs`?